pal/image/native/image-data.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ Copyright (c) 2023 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+import { BaseImageData } from '../base-image-data';
+import type { ImageSource, IMemoryImageSource } from '../types';
+import { ccwindow } from '../../../cocos/core/global-exports';
+import { getError } from '../../../cocos/core';
+
+export class ImageData extends BaseImageData {
+    public destroy (): void {
+        if (this.data && this.data instanceof HTMLImageElement) {
+            // JSB element should destroy native data.
+            // TODO: Property 'destroy' does not exist on type 'HTMLImageElement'.
+            // maybe we need a higher level implementation called `pal/image`, we provide `destroy` interface here.
+            // issue: https://github.com/cocos/cocos-engine/issues/14646
+            (this.data as any).destroy();
+        }
+        super.destroy();
+    }
+    public rawData (): unknown {
+        if (this._imageSource instanceof HTMLCanvasElement) {
+            // @ts-ignore
+            return this._imageSource._data.data;
+        } else if (this._imageSource instanceof HTMLImageElement) {
+            // @ts-ignore
+            return this._imageSource._data;
```

Comment:
we don't recommend using ts-expect-error or ts-ignore, please follow the [codeing-style-guidance](https://github.com/cocos/3d-tasks/blob/master/workflows/coding-style-guidance.md#%E7%A6%81%E7%94%A8-ts-expect-error)

Reply:
Okay, I've removed.